palo_alto_cortex_xdr/plugin.spec.yaml

```diff
@@ -0,0 +1,295 @@
+        type: string
+        description: "Operational status description"
+        required: false
+    os_type:
+        title: "OS Type"
+        type: string
+        description: "OS type"
+        required: false
+    scan_status:
+        title: "Scan Status"
+        type: string
+        description: "Scan status"
+        required: false
+    users:
+        title: "Users"
+        type: "[]string"
+        description: "Users"
+        required: false
+
+connection:
+  fqdn:
+    title: FQDN
+    description: Fully qualified domain name
+    type: string
+    required: true
+    example: https://api-example.xdr.us.paloaltonetworks.com/
+  api_key_id:
+    title: API Key ID
+    description: The API Key ID shown in the API Keys table in settings. e.g. 1, 2, 3
+    type: int
+    required: true
+    example: 1
+  api_key:
+    title: API Key
+    description: The API Key that is generated when creating a new key
+    type: credential_secret_key
+    required: true
+    example: '1234123412341234asdfasdfasdfasdfasdf1234123412341234123412341234asdfasdfasdfasdfasdf123412341234123412341234asdfasdfasdfasdfasdf'
+  security_level:
+    title: Security Level
+    description: The Security Level of the key provided. This can be found in the API Key settings table in the Cortex XDR settings
+    type: string
+    required: true
+    example: Standard
+    default: Standard
+    enum:
+      - Advanced
+      - Standard
+actions:
+  get_endpoint_details:
+    title: Get Endpoint Details
+    description: Get information about an endpoint
+    input:
+      endpoint:
+        title: Endpoint
+        description: The endpoint to get information about. Endpoint will accept IPv4, hostnames, and endpoint IDs
+        type: string
+        required: true
+        example: 0123456abcdef12345abcde12345abcd
+    output:
+      total_count:
+        title: Total Count
+        description: Number of results found (max 100)
+        type: integer
+        required: true
+      endpoints:
+        title: Endpoints
+        description: Any endpoints that match the given endpoint information
+        type: "[]endpoint"
+        required: true
+  isolate_endpoint:
+    title: Isolate Endpoint
+    description: Isolate an endpoint
+    input:
+      endpoint:
+        title: Endpoint
+        description: Endpoint to take isolation action on. This can be IPv4, hostnames, and endpoint IDs
+        type: string
+        required: true
+        example: 0123456abcdef12345abcde12345abcd
+      isolation_state:
+        title: Isolation State
+        description: True to isolate host, false to unisolate a host
+        type: string
+        required: true
+        default: Isolate
+        example: Unisolate
+        enum:
+          - Isolate
+          - Unisolate
+      whitelist:
+        title: Whitelist
+        description: This list contains a set of devices that should not be blocked. This can include IPs, hostnames, and device IDs
```

Comment:
```suggestion
        description: This list contains a set of devices that should not be blocked. This can be a combination of IPv4 addresses, hostnames, or endpoint IDs
```